src/core/agent/services/utils.ts

```diff
@@ -27,4 +28,34 @@ function getCredentialShortDetails(
   };
 }
 
+export const OnlineOnly = (
+  target: any,
+  propertyKey: string,
+  descriptor: PropertyDescriptor
+) => {
+  const originalMethod = descriptor.value;
+  descriptor.value = async function (...args: any[]) {
+    const isKeriOnline = Agent.agent.getKeriaOnlineStatus();
+    if (!isKeriOnline) {
+      throw new Error(Agent.KERIA_CONNECTION_BROKEN);
+    }
+    // Call the original method
+    try {
+      const executeResult = await originalMethod.apply(this, args);
+      return executeResult;
+    } catch (error) {
+      const errorStack = (error as Error).stack;
+      /** If the error is failed to fetch with signify,
+       * we retry until the connection is secured*/
+      if (errorStack && /SignifyClient/gi.test(errorStack)) {
+        await Agent.agent.bootAndConnect(1000);
+        const executeResult = await originalMethod.apply(this, args);
+        return executeResult;
```

Comment:
Hmm, so when we are online and the connection goes: we will try to call Signify and then do bootAndConnect if we can't.

The only problem (and apologies for not noticing until now) is we will be hang here until it reconnects.

Instead, we should remove the await from bootAndConnect (so its in the background), and just throw `KERIA_CONNECTION_BROKEN`. So the UI can show an error and then they will see the connection be restored with a toast or whatever.